frontend/src/components/AuthForms.tsx

```diff
@@ -0,0 +1,286 @@
+        <TabsList className="grid w-full grid-cols-2">
+          <TabsTrigger value="login">Login</TabsTrigger>
+          <TabsTrigger value="signup">Signup</TabsTrigger>
+        </TabsList>
+        <TabsContent value="login">
+          <Card>
+            <CardHeader>
+              <CardTitle>Login</CardTitle>
+              <CardDescription>
+                Enter your username and password to login.
+              </CardDescription>
+              {loginError && (
+                <p className="text-sm text-red-500">{loginError}</p>
+              )}
+            </CardHeader>
+            <form onSubmit={handleSubmitLogin(onSubmitLogin)}>
+              <CardContent className="space-y-2">
+                <div className="space-y-1">
+                  <Label htmlFor="login-username">Username</Label>
+                  <Input
+                    id="login-username"
+                    type="text"
+                    {...registerLogin("username")}
+                    onChange={handleLoginInputChange}
+                  />
+                  {loginErrors.username && (
+                    <p className="text-sm text-red-500">
+                      {loginErrors.username.message}
+                    </p>
+                  )}
+                </div>
+                <div className="space-y-1">
+                  <Label htmlFor="login-password">Password</Label>
+                  <div className="relative">
+                    <Input
+                      id="login-password"
+                      type={showLoginPassword ? "text" : "password"}
+                      {...registerLogin("pass")}
+                      onChange={handleLoginInputChange}
+                    />
+                    <Button
+                      type="button"
+                      variant="ghost"
+                      size="icon"
+                      className="absolute right-0 top-0 h-full px-3 py-2 hover:bg-transparent"
+                      onClick={() => setShowLoginPassword(!showLoginPassword)}
+                    >
+                      {showLoginPassword ? (
+                        <EyeOff className="h-4 w-4 text-gray-500" />
+                      ) : (
+                        <Eye className="h-4 w-4 text-gray-500" />
+                      )}
+                    </Button>
+                  </div>
+                  {loginErrors.pass && (
+                    <p className="text-sm text-red-500">
+                      {loginErrors.pass.message}
+                    </p>
+                  )}
+                </div>
+              </CardContent>
+              <CardFooter>
+                <Button
+                  type="submit"
+                  className="w-full"
+                  disabled={loginMutation.isPending}
+                >
+                  {loginMutation.isPending ? "Logging in..." : "Login"}
+                </Button>
+              </CardFooter>
+            </form>
+          </Card>
```

Comment:
Put the login card in the separate LoginCard component - it would make this code more readable